src/Mvc/Mvc.DataAnnotations/src/DataAnnotationsMetadataProvider.cs

```diff
@@ -380,5 +405,26 @@ private static string GetDisplayGroup(FieldInfo field)
 
             return string.Empty;
         }
+
+        // Internal for testing
+        internal static bool IsNonNullable(IEnumerable<object> attributes)
+        {
+            // [Nullable] is compiler synthesized, comparing by name. 
```

Comment:
Let's keep an eye on https://github.com/dotnet/corefx/issues/36222 in case they start exposing it

Reply:
I would certainly prefer that, but this code won't really need to change, it will still be correct.